proposals/text/3875-table-span.md

```diff
@@ -0,0 +1,228 @@
+prediction = reader.predict(query="Who was in the most number of movies?", documents=[table_doc])
+answer = prediction["answers"][0]
+
+# New feature
+print(answer.context.iloc[answer.offsets_in_context[0].row, answer.offsets_in_context[0].col])
+```
+
+# Motivation
+
+## Why do we need this feature?
+To allow users to easily look up the answer cell in the returned table to fetch the answer text
+directly from the table, identify the row or column labels for that answer, or generally perform operations on the table
+near or around the answer cell.
+
+Currently, we return the location of the answer in the **linearized** version of the table, so we can use the
+`Span` dataclass. The `Span` dataclass is reproduced below:
+```python
+@dataclass
+class Span:
+    start: int
+    end: int
+    """
+    Defining a sequence of characters (Text span) or cells (Table span) via start and end index.
+    For extractive QA: Character where answer starts/ends
+    For TableQA: Cell where the answer starts/ends (counted from top left to bottom right of table)
+
+    :param start: Position where the span starts
+    :param end:  Position where the span ends
+    """
+```
+This is inconvenient for users because they would need to know how the table is linearized (column major or row major)
+so they could reconstruct the column and row indices of the answer before they could locate the answer cell in the table.
+
+## What use cases does it support?
+Some examples are already stated above but to recap, to easily perform operations on the table near or around the answer cell.
+
+## What's the expected outcome?
+The addition of a new dataclass called `TableSpan` that would look like
+```python
+@dataclass
+class TableSpan:
+    col: int
+    row: int
+    """
+    Defining a table cell via the column and row index.
+
+    :param col: Column index of the cell
+    :param row: Row index of the cell
+    """
+```
+**Note:** I am open to a name change since this isn't really a span, but the location of a single table cell.
+
+# Detailed design
+
+**New terminology:** `TableSpan` or something similar (e.g. `Cell`,`CellLoc`, etc.). The new name for the dataclass to
+store the column and row index of the answer cell.
+
+**Basic Example:** [Above Basic Example](#basic-example)
+
+## Code changes
+- Addition of `TableSpan` dataclass to https://github.com/deepset-ai/haystack/blob/main/haystack/schema.py
+```python
+@dataclass
+class TableSpan:
+    col: int
+    row: int
+    """
+    Defining a table cell via the column and row index.
+
+    :param col: Column index of the cell
+    :param row: Row index of the cell
+    """
+```
+
+- Updating code (e.g. schema objects, classes, functions) that use `Span` to also support `TableSpan` where appropriate.
+This includes:
+- Updating the `Answer` dataclass to support `TableSpan` as a valid type for `offsets_in_document` and `offsets_in_context`
```

Comment:
I would also like this approach. And I agree it would help with validation. Using this approach we could also remove the Dataframe as a valid context for `Answer` and only make it valid for `TableAnswer`. 

If others agree I can update the proposal with this being the main suggestion.

Reply:
Would that make it more difficult to users to join the results of a TableReader and a FARMReader? If `TableAnswer` is a subclass of `Answer` that should work without problems, right? Please keep this use case in mind too. 
I think it's a good idea. What I am concerned about  is that we handle `Document` and `Answer` inconsistently then:
On the one side, there is one main `Document` class with `content_type: One of "text", "table", "image" or "audio"` (and there is `SpeechDocument`). On the other side, we have `Answer` and `SpeechAnswer` and with this proposal we might add `TableAnswer`. With the new ImageToText node we are not far from an `ImageAnswer` (text-based answer that also contains some accessory image information). That's quite some subclasses and it's a bigger decision whether we want to go in this direction.

Reply:
That's a great point @julian-risch. It sounds like we should have a proposal/discussion on the architecture we would like to use for handling the different `Answer` and `Document` types we have for each of the different `content_types` we support. Then we could make this behavior consistent across the different `content_types`. 

For this proposal, it may be safer then to not create `TableAnswer` and wait for the refactor discussion to decide whether we should be creating more subclasses. What do you think? @julian-risch @ZanSara

Reply:
Sounds like a good plan to me, yes. If we keep this proposal here a bit smaller it allows us to make faster progress. 👍

Reply:
Should I open an issue to remind us to open a proposal on this subclassing topic of `Document` and `Answer`?

Reply:
Opened issue for the reminder [here](https://github.com/deepset-ai/haystack/issues/3975)